src/sentry/integrations/slack/message_builder/notifications.py

```diff
@@ -0,0 +1,32 @@
+import re
+from urllib.parse import urljoin
+
+from sentry.integrations.slack.utils import LEVEL_TO_COLOR
+
+
+def build_notification_footer(links, notification_type):
+    referrer = re.sub("Notification$", "Slack", notification_type)
+    settings_url = urljoin(links["settings_url"], "?referrer=" + referrer)
+
+    if notification_type == "ReleaseActivityNotification":
+        # groups are not associated with a deploy notification
+        # so in this one case, the footer is different
+        return f"<{settings_url}|Notification Settings>"
+
+    group_url = urljoin(links["group_url"], "?referrer=" + referrer)
+    short_id = links["short_id"]
+    return f"<{group_url}|{short_id}> via <{settings_url}|Notification Settings>"
+
+
+def build_notification_attachment(notification):
+    links = notification.get_dm_links()
+    notification_type = notification.__class__.__name__
+    footer = build_notification_footer(links, notification_type)
```

Comment:
I think you should change `build_notification_footer` to take the `ActivityNotification` object and internally calculate `links`.